### PR TITLE
Login api changes for com-api envelope

### PR DESCRIPTION
**Pull request by UmeshBhosale**

### CODE REVIEW

File: users/users/login.php [hunk 1]
```diff
@@ -53,6 +53,12 @@ public function keygen()
 			$id = $model->getUserId('email', $username);            
 		}
 
+		$res					=	array();
```
Comment (twsvaishali) [inline users/users/login.php]: @UmeshBhosale Please go through following document. The response should be same as mentioned here.
http://git.tekdi.net/techjoomla/shika/wikis/rest-api-users

Also, we will also need to do changes in com_api to handle validation errors.

File: users/users/login.php [hunk 2]
```diff
@@ -100,13 +100,13 @@ public function keygen()
 		
 		if( !empty($key) )
 		{
-			$dataObj->results = $key;
-			$dataObj->id = $id;
+			$dataObj['results'] -> tocken 	= $key;
```
Comment (twsvaishali) [inline users/users/login.php]: tocken is wrong.. it should be token

File: users/users/login.php [hunk 3]
```diff
@@ -54,9 +54,9 @@ public function keygen()
 		}
 
 		$res					=	array();
```
Comment (twsvaishali) [inline users/users/login.php]: $result should be array or object? $result = new stdClass;

File: users/users/login.php (hunk 1 above)
Comment (twsvaishali) [inline users/users/login.php]: @UmeshBhosale  In the starting of the file, following list of imports is written.

jimport('joomla.plugin.plugin');
jimport('joomla.html.html');
jimport('joomla.application.component.controller');
jimport('joomla.application.component.model');
jimport('joomla.user.helper');
jimport('joomla.user.user');
jimport('joomla.application.component.helper');

I dont thinnk all of these are needed

File: users/users/login.php [hunk 4]
```diff
@@ -53,6 +53,12 @@ public function keygen()
 			$id = $model->getUserId('email', $username);            
 		}
 
+		$res					=	array();
+		$dataObj['results']		=	new stdClass;
```
Comment (twsvaishali) [inline users/users/login.php]: @UmeshBhosale  please check code again.. $dataObj is object or array?

File: users/users/login.php [hunk 5]
```diff
@@ -94,17 +100,15 @@ public function keygen()
 		
 		if( !empty($key) )
 		{
-			$obj->auth = $key;
-			$obj->code = '200';
-			//$obj->id = $user->id;
-			$obj->id = $id;
+			$dataObj['results'] -> token 	= $key;
```
Comment (twsvaishali) [inline users/users/login.php]: Also check this. Changing $dataObj later, will change the $result?

Comment (UmeshBhosale) [inline users/users/login.php]: Yes, It will update $res object

Comment (UmeshBhosale) [inline users/users/login.php]: @twsvaishali I have updated MR. Please review.

**Pull request closed without merging**